src/smolagents/remote_executors.py

```diff
@@ -245,7 +245,7 @@ def run_code_raise_errors(self, code_action: str, return_final_answer: bool = Fa
         """
         try:
             if return_final_answer:
-                match = self.final_answer_pattern.match(code_action)
+                match = self.final_answer_pattern.search(code_action)
                 if match:
```

Comment:
The reason of the failure you still have in the test is wrapped_code doesn't contain the original code before final_answer. Here is the fixed version:
if match:
                    result_expr = match.group(1)
                    code_before_final_answer = ""
                    for line in code_action.strip().split("\n")[:-1]:
                        code_before_final_answer += line + "\n"
                    wrapped_code = f"""{code_before_final_answer}
import pickle, base64
_result = {result_expr}
print("RESULT_PICKLE:" + base64.b64encode(pickle.dumps(_result)).decode())
"""

Reply:
Thanks, @maeste! This will fix `test_execute_image_output`.